Ginger/Ginger/ApplicationModelsLib/APIModels/APIModelWizard/AddAPIModelSelectTypePage.xaml.cs

```diff
@@ -388,20 +400,170 @@ private void BrowseForTemplateFiles(string files = null)
             {
                 foreach (String file in dlg.FileNames)
                 {
-                    AddAPIModelWizard.XTFList.Add(new TemplateFile() { FilePath = file });
+                    if (await ValidateFile(file))
+                    {
+                        AddAPIModelWizard.XTFList.Add(new TemplateFile() { FilePath = file });
+                    }
+                }
+            }
+        }
+
+        private async Task<bool> ValidateFile(string fileName = "")
+        {
+            bool bIsFileValid = false;
+            try
+            {
+                if (APITypeComboBox.SelectedValue.ToString() == eAPIType.WSDL.ToString())
+                {
+                    fileName = xURLTextBox.Text;
+                    bIsFileValid = await LoadWSDLFileValidation();
+                }
+                else if (APITypeComboBox.SelectedValue.ToString() == eAPIType.Swagger.ToString())
+                {
+                    SwaggerParser swagger = new SwaggerParser();
+                    ObservableList<ApplicationAPIModel> swaggerList = new ObservableList<ApplicationAPIModel>();
+                    swaggerList = swagger.ParseDocument(fileName, swaggerList);
+                    if (swaggerList == null || swaggerList.Count == 0)
+                    {
+                        bIsFileValid = false;
+                    }
+                    else
+                    {
+                        bIsFileValid = true;
+                    }
+                }
+                else if (APITypeComboBox.SelectedValue.ToString() == eAPIType.JsonTemplate.ToString())
+                {
+                    JSONTemplateParser jsonParser = new JSONTemplateParser();
+                    ObservableList<ApplicationAPIModel> jsonList = new ObservableList<ApplicationAPIModel>();
+
+                    jsonList = jsonParser.ParseDocument(fileName, jsonList);
```

Comment:
Please see if we can use function which you have implemented for file validation